### ISSUE TITLE
Migrate some automatically generated scenario's (and protect them)

### ISSUE THREAD

**Issue by DorinevanderVlies** [quintel/etmodel#3035]
I the last year we have used ETM-tools to automatically generate scenario's on several occasions. These scenario's are currently not protected and are therefor not part of migrations. For some of them we gave the scenario numbers to clients for future reference. 
[Overview: unprotected scenarios for migration.xlsx](https://github.com/quintel/etmodel/files/3182341/unprotected.scenarios.for.migration.xlsx)

* Is it possible to rerun the relevant migrations for these scenario's?
* Is it possible to protect these scenario's so they are part of future migrations?
* Will these extra scenario's (~ 1000) slow down future scenario's? If so, the scenario's in the tab "OPeningsbod_Stedin_0.8 do not have to be taken into account

More general question:
* How should we deal with this in the future?

**Comment by antw** [comment]
> Is it possible to rerun the relevant migrations for these scenario's?

It's a bit of a pain, as we have to _duplicate_ the migrations and run them for these specific scenarios, but it can be done.

> Is it possible to protect these scenario's so they are part of future migrations?

Yes.

> Will these extra scenario's (~ 1000) slow down future scenario's? If so, the scenario's in the tab "OPeningsbod_Stedin_0.8 do not have to be taken into account

It will slow down future migrations a little, but I'm not too worried about that. If you want the Stedin scenarios to be kept up-to-date, we'll have to migrate and protect them.

> How should we deal with this in the future?

I think the etm-tools script(s) used to create these scenarios should have an option to protect the scenarios when they're created. I'm guessing [cp-scenario](https://github.com/quintel/etm-tools/tree/master/cp-scenario) was the tool which was used?

**Comment by DorinevanderVlies** [comment]
> It's a bit of a pain, as we have to duplicate the migrations and run them for these specific scenarios, but it can be done.

What is your time estimation?

> I think the etm-tools script(s) used to create these scenarios should have an option to protect the scenarios when they're created. I'm guessing cp-scenario was the tool which was used?

Sounds good!
Yes, we used a combination of  [cp-scenario](https://github.com/quintel/etm-tools/tree/master/cp-scenario) and [csv-update-scenarios](https://github.com/quintel/etm-tools/tree/master/csv-update-scenarios)

**Comment by antw** [comment]
I've pushed these migrations to the openingsbod-migrations branch.

> the scenario's in the tab "OPeningsbod_Stedin_0.8 do not have to be taken into account

I did not include these scenarios, since "0.8" suggested that they might not be the finished versions? If I'm wrong about that, and you wish for them to be migrated and protected, it's trivial for me to add them before the branch is merged. As soon as you let me know how you want to proceed with the Stedin scenarios I can send the migrations to production.

> I think the etm-tools script(s) used to create these scenarios should have an option to protect the scenarios when they're created. 

This has been added. [cp-scenario now accepts an optional `--protect` argument](https://github.com/quintel/etm-tools/tree/master/cp-scenario#protected-scenarios) which will mark the created scenarios as protected.
